lib/errorHandlers/index.ts

```diff
@@ -82,21 +75,37 @@ function debugError(error, context = {}) {
   }
 }
 
-class ApiErrorContext {
-  constructor(props = {}) {
-    /** @type {number} */
+export class ApiErrorContext {
+  accountId?: number;
+  request?: string;
+  payload?: string;
+  projectName?: string;
+
+  constructor(
+    props: {
+      accountId?: number;
+      request?: string;
+      payload?: string;
+      projectName?: string;
+    } = {}
+  ) {
     this.accountId = props.accountId;
-    /** @type {string} */
     this.request = props.request || '';
-    /** @type {string} */
     this.payload = props.payload || '';
-    /** @type {string} */
     this.projectName = props.projectName || '';
   }
 }
 
-module.exports = {
-  logError,
-  debugError,
-  ApiErrorContext,
-};
+function isErrorWithCause(error: unknown): error is { cause: unknown } {
+  return typeof error === 'object' && error !== null && 'cause' in error;
+}
```

Comment:
The base `Error` class should have a `cause`, so we can potentially make this a more general `isError` predicate that looks something like:
```typescript
function isError(error: unknown): error is Error {
  return error instanceof Error;
}
```

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1251/commits/9fc06a16fccfda6b8e5d060b6c4703e0a83a0d07